### ISSUE TITLE
Feature request: vnstat like traffic monitor

### ISSUE THREAD

**Issue by coreshift** [passteque/gluetun#251]
1. What's the feature?
Ability to find out how much data has been sent/received through the vpn tunnel and keeping track of daily/monthly totals.

2. Why do you need this feature?
Limited data plans from ISP

3. Extra information?
Maybe there's already a way to find this out without adding a new feature?

**Comment by qdm12** [comment]
I can add it. For now you could shell in the container and check `iptables -nvL` which would show how much bytes went through the firewall. But anyway, that's a good idea, I'll add it as an http endpoint on the http control server. One day I'll make a UI too which should help out monitor that. There is also #131 such that you could receive mobile notifications, I'll keep this in mind such that you can set warning limits.

**Comment by coreshift** [comment]
Thank you. Will the data persist across server reboots or container restarts? I'm sure it would make things more complicated if bw data has to be periodically stored to a data file or something. I tend to reboot or restart every few days.

**Comment by qdm12** [comment]
Yes I can make it persistent to a file in `/gluetun` relatively easily, good idea too.

**Comment by qdm12** [comment]
For my future self, `/proc/net/dev` seems to contain the data required. `/proc/net/ip_vs_stats` also contains current network rates. It might be interesting also to export these as prometheus metrics.

**Comment by TheGameProfi** [comment]
Hey everyone 👋

Since there hasn’t been much activity or communication lately, I went ahead and created a Prometheus Exporter myself.
It’s still a work in progress — I’m currently experimenting with getting it to auto-configure, but the core functionality is already there.

Feel free to check it out and let me know what you think:
👉 [Gluetun-Exporter](https://github.com/thecfu/gluetun-exporter)

I’m totally open to feedback, ideas, or suggestions! 😊

**Comment by qdm12** [comment]
See #3421

**Comment by qdm12** [comment]
See https://github.com/qdm12/gluetun-wiki/blob/latest/setup/advanced/metrics.md
